kube-runtime/src/reflector/object_ref.rs

```diff
@@ -69,32 +56,41 @@ pub struct Extra {
     pub uid: Option<String>,
 }
 
-impl<K: Resource> ObjectRef<K>
-where
-    K::DynamicType: Default,
-{
+impl ObjectRef {
+    /// Create a blank `ObjectRef` with a name
     #[must_use]
     pub fn new(name: &str) -> Self {
-        Self::new_with(name, Default::default())
+        Self {
+            name: name.into(),
+            namespace: None,
+            types: None,
+            extra: Extra::default(),
+        }
     }
 
+    /// Creates `ObjectRef` from the resource
     #[must_use]
-    pub fn from_obj(obj: &K) -> Self
-    where
-        K: Resource,
-    {
-        Self::from_obj_with(obj, Default::default())
+    pub fn from_obj<K: TypeInfo>(obj: &K) -> Self {
+        let meta = obj.meta();
+        Self {
+            name: meta.name.clone().unwrap_or_default(),
+            namespace: meta.namespace.clone(),
+            types: obj.types(),
```

Comment:
Not super happy about each `ObjectRef` needing its own clone of all the type metadata when we know it all statically anyway.

Reply:
This is a lot less cloning in the dynamic case though. Before we had to clone the full dynamictype, which is a full `ApiResource`. I don't see a way to be able to get type information out of dynamic types without either:

- the current complicated dynamic type feeding (which is more clone heavy on dyn types, and has more difficult ctors)
- actually cloning the typemeta for each object

I don't think it's such a big hit. It's worth it to simplify the api if we can start moving away from `_with` suffixed ctors. Our reflectors are the real memory hog here that's worth optimising for imo.